pokemongo_bot/cell_workers/catch_lured_pokemon.py

```diff
@@ -12,39 +13,52 @@ class CatchLuredPokemon(BaseTask):
     def work(self):
         lured_pokemon = self.get_lured_pokemon()
         if lured_pokemon:
-            self.catch_pokemon(lured_pokemon)
+            for pokemon in lured_pokemon:
+                self.catch_pokemon(pokemon)
 
     def get_lured_pokemon(self):
+        forts_in_range = []
+        pokemon_to_catch = []
         forts = self.bot.get_forts(order_by_distance=True)
 
         if len(forts) == 0:
             return False
 
-        fort = forts[0]
-        details = fort_details(self.bot, fort_id=fort['id'],
```

Comment:
What we need is an ORM-like lib to avoid doing any manual call using the raw api object. Otherwise we code lots of duplicated cache, like the one we already have for pokestops timers. Also caching all seen pokestops may have a very big impact in memory usage for long-running bots. A simple cache for all pokestops is not the way, we need something smarter.